diagnostic.js

```diff
@@ -29,26 +29,79 @@ mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
+const db = mongoose.connection;
 
-const create = (name, description, startYear, endYear) => {};
+const Movement = require('./models/movement.js');
+
+const done = function() {
+  db.close();
+};
+
+const create = (name, description, startYear, endYear, ancestor) => {
+  Movement.create({
+    name: name,
+    description: description,
+    startYear: startYear,
+    endYear: endYear,
+    ancestor: ancestor
+  }).then(function(movement) {
+    console.log(movement.toJSON());
+    done();
+  }).catch(console.error).then(done);
+};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  let searchParams = {};
+if (arguments[0] && arguments[1]) {
+  let field = arguments[0];
+  let criterion = arguments[1];
+  if (criterion[0] === '/') {
+    let regex = new RegExp(criterion.slice(1, criterion.length - 1));
+    searchParams[field] = regex;
+  } else {
+    searchParams[field] = criterion;
+  }
+}
+
+  Movement.find({searchParams}).then(function(movements){
+    movements.forEach(function(movement){
+      console.log(movement.toJSON());
+    });
+  }).catch(console.error).then(done);
+};
+
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id).then(function(movement){
+    console.log(movement.toJSON());
```

Comment:
Don't forget to check if `movement` is `null`! That's what the result will be if no Movement is found.